src/game_controller.py

```diff
@@ -0,0 +1,99 @@
+import os
+import threading
+import time
+
+import keyboard
+from cv2 import cv2
+
+from bot import Bot
+from config import Config
+from death_manager import DeathManager
+from game_recovery import GameRecovery
+from game_stats import GameStats
+from health_manager import HealthManager
+from logger import Logger
+from messenger import Messenger
+from screen import Screen
+from utils.misc import kill_thread
+
+
+class GameController:
+
+    def __init__(self):
+        self.screen = None
+        self.health_manager = None
+        self.death_manager = None
+        self.game_recovery = None
+        self.game_stats = None
+
+    def run_bot(self, config: Config, pick_corpse: bool = False):
+        keyboard.remove_hotkey(config.general['resume_key'])
+        # Create global variables if not done yet (id est: first start)
+        if not self.screen:
+            self.screen = Screen(config.general["monitor"])
+        if not self.health_manager:
+            # Run health monitor thread
+            self.health_manager = HealthManager(self.screen)
+            health_monitor_thread = threading.Thread(target=self.health_manager.start_monitor)
+            health_monitor_thread.daemon = True
+            health_monitor_thread.start()
+        if not self.death_manager:
+            # Run death monitor thread
+            self.death_manager = DeathManager(self.screen)
+            death_monitor_thread = threading.Thread(target=self.death_manager.start_monitor)
+            death_monitor_thread.daemon = True
+            death_monitor_thread.start()
+        if not self.game_recovery:
+            self.game_recovery = GameRecovery(self.screen, self.death_manager)
+        if not self.game_stats:
+            self.game_stats = GameStats()
```

Comment:
All the object creation could be done in the __init__() without these if clauses, no? Adding the config as member var and ditching it from the run_bot() method.

Reply:
yeah very fair point I did it actually before reading your comment, this was realyl part of mid changes I made to transition the logic into the new classes and should definetely go in the constructor